src/routes/formsgSiteLaunch.ts

```diff
@@ -0,0 +1,212 @@
+
+    const subDomainSettings = [
+      {
+        branchName: "master",
+        prefix: "",
+      },
+    ]
+
+    if (redirectionDomain) {
+      subDomainSettings.push({
+        branchName: "master",
+        prefix: "www",
+      })
+    }
+
+    logger.info(
+      `Launch site form submission [${submissionId}] (repoName '${repoName}', domain '${primaryDomain}') requested by <${requesterEmail}>`
+    )
+
+    // 2. Check arguments
+    if (!requesterEmail) {
+      // Most errors are handled by sending an email to the requester, so we can't recover from this.
+      throw new BadRequestError(
+        "Required 'Government E-mail' input was not found"
+      )
+    }
+
+    if (!agencyEmail) {
+      // Most errors are handled by sending an email to the requester, so we can't recover from this.
+      throw new BadRequestError("Required 'Agency E-mail' input was not found")
+    }
+
+    try {
+      if (!primaryDomain) {
+        const err = `A primary domain is required`
+        await this.sendLaunchError(
+          [requesterEmail, agencyEmail],
+          repoName,
+          submissionId,
+          err
+        )
+        return res.sendStatus(200)
+      }
+      if (!repoName) {
+        const err = `A repository name is required`
+        await this.sendLaunchError(
+          [requesterEmail, agencyEmail],
+          repoName,
+          submissionId,
+          err
+        )
+        return res.sendStatus(200)
+      }
+
+      const agencyUser = await this.usersService.findByEmail(agencyEmail)
+      const requesterUser = await this.usersService.findByEmail(requesterEmail)
+
+      if (!agencyUser) {
+        const err = `Form submitter ${agencyEmail} is not an Isomer user. Register an account for this user and try again.`
+        await this.sendLaunchError(
+          [requesterEmail, agencyEmail],
+          repoName,
+          submissionId,
+          err
+        )
+        return res.sendStatus(200)
+      }
+
+      if (!requesterUser) {
+        const err = `Form submitter ${requesterUser} is not an Isomer user. Register an account for this user and try again.`
+        await this.sendLaunchError(
+          [requesterEmail, agencyEmail],
+          repoName,
+          submissionId,
+          err
+        )
+        return res.sendStatus(200)
+      }
+
+      // 3. Use service to Launch site
+      // note: this function is not be async due to the timeout for http requests.
+      const launchSite = this.infraService.launchSite(
+        submissionId,
+        requesterUser,
+        agencyUser,
+        repoName,
+        primaryDomain,
+        subDomainSettings
+      )
+
+      // only send success message after promise has been resolved
+      launchSite.then(async () => {
+        await this.sendLaunchSuccess(
+          // fields below are guarenteed to be a string due to prior checks
+          <string>requesterEmail,
+          <string>repoName,
+          submissionId
+        )
+      })
```

Comment:
fixed 51333a7. Am not using the typical `await` as `launchSite` takes longer than 60 seconds to return a response, and this might lead to HTTP timeout errors.

Reply:
double checking here - we only ever return `200 OK` as this is a webhook endpoint to receive form submissions from formsg.

this implies that we can send the appropriate on finishing message processing, as that's all the status code will indicate to forms (see [here](https://github.com/opengovsg/FormSG/blob/9db3249a70c9167f915f4c2ced9b1030f4d3e561/src/app/modules/webhook/webhook.service.ts#L97) - once message delivery is successful, they are technically absolved). 

should we just shift the `res.send` up to when we finish processing and carry on with this thereafter?

Reply:
this makes sense - 6ba04fd